### PR TITLE
Connext merge

### PR DESCRIPTION
**Pull request by adamsj-ros**
This is the update to pull in all the latest **eloquent** changes/branch from `rosidl_typesupport_connext_cpp`.  `rmw_opendds` has an associated change/PR to match new names.

### CODE REVIEW

File: rosidl_typesupport_opendds_cpp/RawData.idl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const long KEY_HASH_LENGTH_16 = 16;
+
+struct RawData {
+	@key octet       key_hash[KEY_HASH_LENGTH_16];
+	sequence<octet>  key_data;
+	sequence<octet>  raw_data;
```
Comment (iguessthislldo) [inline rosidl_typesupport_opendds_cpp/RawData.idl]: OpenDDS doesn't support anonymous types at the moment.

Comment (adamsj-ros) [reply rosidl_typesupport_opendds_cpp/RawData.idl]: @kuznetsovmoci How do you want to handle this situation? It's what Connext has in their code but not supported in OpenDDS.

Comment (iguessthislldo) [reply rosidl_typesupport_opendds_cpp/RawData.idl]: Well for sequences of primitive types like `octet` you need to use some predefined types in TAO. Also you will need to add `@topic` to the struct or change the options on `opendds_idl` to make all types topic types by default.

```
#include <tao/OctetSeq.pidl>

const long KEY_HASH_LENGTH_16 = 16;

@topic
struct RawData {
	@key octet key_hash[KEY_HASH_LENGTH_16];
        OctetSeq key_data;
	OctetSeq raw_data;
};
```

**Comment by mitza-oci** [comment]
We already solved this

File: rosidl_typesupport_opendds_cpp/RawData.idl (hunk 1 above)
Comment (adamsj-ros) [reply rosidl_typesupport_opendds_cpp/RawData.idl]: I updated based on what @mitza-oci  and I did for opendds_static_serialized_data.idl.

**Review by kuznetsovmoci — COMMENTED** [review]
rosidl_typesupport_opendds_cpp file used to be in rosidl_typesupport_opendds_cpp folder and its content is different. Now it is in the bin folder. Should we remove the old one?

**Review by kuznetsovmoci — APPROVED** [review]

**Comment by adamsj-ros** [comment]
> rosidl_typesupport_opendds_cpp file used to be in rosidl_typesupport_opendds_cpp folder and its content is different. Now it is in the bin folder. Should we remove the old one?

The files are different. Let's leave them as they are for now.

**Pull request merged**